### PR TITLE
Set description via advapi32

### LINKED ISSUE #208: WinSW should use advapi to write service description
Follow-up to #183. Currently WinSW writes description directly to the registry, which is not guaranteed to work reliably on all Windows versions.

### PR DESCRIPTION
**Pull request by NextTurn**
Fixes #208 

Hold for #362

### CODE REVIEW

File: src/Core/ServiceWrapper/Main.cs [hunk 1]
```diff
@@ -642,39 +642,21 @@ public static void Run(string[] _args, ServiceDescriptor? descriptor = null)
                     password,
                     descriptor.ServiceDependencies);
 
-                // update the description
-                /* Somehow this doesn't work, even though it doesn't report an error
-                Win32Service s = svc.Select(d.Id);
-                s.Description = d.Description;
```
Comment (oleg-nenashev) [inline src/Core/ServiceWrapper/Main.cs]: I wonder whether it is still an issue on old systems, but I am happy to remove this hack

Comment (NextTurn) [reply src/Core/ServiceWrapper/Main.cs]: Do you mean setting description via `Win32_Service`?

Comment (oleg-nenashev) [reply src/Core/ServiceWrapper/Main.cs]: Yes. I have no practical way to test on .NET Framework 2 now. But we can go ahead and merge it. It is not a critical functionality, so we can restore this hack for old platforms if needed

Comment (NextTurn) [reply src/Core/ServiceWrapper/Main.cs]: The `Win32_Service` approach doesn't work because `Description` is a read-only property.

**Comment by oleg-nenashev** [comment]
@NextTurn just to get on the same page w.r.t pull request labeling for the changelogs, do you see a difference for users here? My labeling approach is that `enhancement` goes to user-facing features while the rest goes to `internal`

**Comment by NextTurn** [comment]
> My labeling approach is that `enhancement` goes to user-facing features while the rest goes to `internal`.

Sounds good. This should be `internal` then.

> Probably I should explicitly add descriptions to labels

Completely agreed.

**Comment by NextTurn** [comment]
Is the linked issue (#208) `internal` or not?

**Comment by oleg-nenashev** [comment]
I would say it is internal. There is a "Currently WinSW writes description directly to the registry, which is not guaranteed to work reliably on all Windows versions." note in the ticket, but so far we had no reports confirming it is a case on modern platforms

**Pull request merged**